tests/unit-tests/admin/test-class-sensei-onboarding-api.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Sensei Onboarding REST API tests
+ *
+ * @package sensei-lms
+ * @since   3.1.0
+ */
+
+/**
+ * Class for Sensei_Onboarding API tests.
+ */
+class Sensei_Onboarding_API_Test extends WP_Test_REST_TestCase {
+
+	/**
+	 * A server instance that we use in tests to dispatch requests.
+	 *
+	 * @var WP_REST_Server $server
+	 */
+	protected $server;
+
+	/**
+	 * Test specific setup.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		global $wp_rest_server;
+		$wp_rest_server = new WP_REST_Server();
+		$this->server   = $wp_rest_server;
+
+		do_action( 'rest_api_init' );
+
+	}
+
+	/**
+	 * Test specific teardown.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		global $wp_rest_server;
+		$wp_rest_server = null;
+
+		// Restore Usage tracking option.
+		Sensei()->usage_tracking->set_tracking_enabled( true );
+	}
+
+	/**
+	 * Tests if only privileged users can access the Onboarding API.
+	 *
+	 * @covers Sensei_Onboarding::handle_api_request
+	 */
+	public function testOnlyAdminUserCanAccessOnboardingAPI() {
```

Comment:
This should be broken into two tests, because it is testing two things (Teacher cannot access, admin can access).